spyder/plugins/ipythonconsole/widgets/tests/test_kernelconnect.py

```diff
@@ -0,0 +1,298 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright © Spyder Project Contributors
+#
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+# -----------------------------------------------------------------------------
+
+"""Tests for the Existing Kernel Connection widget."""
+
+# Third party imports
+import sys
+import os
+import pytest
+from qtpy.QtCore import Qt
+from qtpy.QtWidgets import QDialogButtonBox
+import time
+
+# Local imports
+from spyder.plugins.ipythonconsole.widgets.kernelconnect import (
+    KernelConnectionDialog
+)
+from spyder.config.main import CONF
+
+
+# =============================================================================
+# Fixtures
+# =============================================================================
+@pytest.fixture
+def connection_dialog_factory(request):
+    """Set up kernel connection dialog."""
+    class DialogFactory(object):
+        def get(self):
+            dialog = KernelConnectionDialog()
+            request.addfinalizer(dialog.close)
+            return dialog
+
+    def teardown():
+        """Clear existing-kernel config and keyring passwords"""
+        CONF.remove_section("existing-kernel")
+
+        try:
+            import keyring
+            keyring.set_password("existing_kernel",
+                                 "ssh_key_passphrase", "")
+            keyring.set_password("existing_kernel",
+                                 "ssh_password", "")
+        except Exception:
+            pass
+
+    request.addfinalizer(teardown)
+    return DialogFactory()
+
+
+# =============================================================================
+# Tests
+# =============================================================================
+def test_kernel_connection_dialog_remembers_input_ssh_passphrase(
+    qtbot, connection_dialog_factory
```

Comment:
Per PEP8 and our convention, hanging indent continuations of ``def`` statements should be indented two levels, i.e. 8 spaces, and the closing paren should be on the same line as the last parameter. Also, I appreciatiate the descriptiveness but this is probably a little much. Trimming some redundant words:

``test_connection_dialog_remembers_passphrase``